docker/docker-compose.1804.50.yaml

```diff
@@ -19,7 +19,7 @@ services:
   integration-tests:
     image: swift-nio:18.04-5.0
     environment:
-      - MAX_ALLOCS_ALLOWED_1000_reqs_1_conn=36750
+      - MAX_ALLOCS_ALLOWED_1000_reqs_1_conn=45750
```

Comment:
I'm raising the allocation limits here because currently, we're in a bad spot: We prepare a whole `[(String, String)]` for the headers to then turn it into a `ByteBuffer` of header blocks that when consumed is converted into `String` again :). Despite this, on macOS I can't see a performance difference with `wrk` but will look into this more.

I started #857 to remove all the complexity in `HTTPHeaders` that should be large obsolete with Swift 5 Strings.

Reply:
`tcpkali` confirms that despite this, we don't seem to be losing out (on macOS):

```
#after
$ tcpkali -em "GET / HTTP/1.1\r\nHost: google.com\r\n\r\n" -r 10000000000000 --latency-marker "HTTP/1.1" 127.0.0.1:8888
Destination: [127.0.0.1]:8888
Interface lo0 address [127.0.0.1]:0
Using interface lo0 to connect to [127.0.0.1]:8888
Ramped up to 1 connections.
Total data sent:     18.2 MiB (19071232 bytes)
Total data received: 25.1 MiB (26270244 bytes)
Bandwidth per channel: 36.239⇅ Mbps (4529.9 kBps)
Aggregate bandwidth: 20.997↓, 15.243↑ Mbps
Packet rate estimate: 49063.5↓, 1325.3↑ (1↓, 35↑ TCP MSS/op)
Message latency at percentiles: 561.1/576.7/580.3 ms (95/99/99.5%)
Test duration: 10.0093 s.

#before
$ tcpkali -em "GET / HTTP/1.1\r\nHost: google.com\r\n\r\n" -r 10000000000000 --latency-marker "HTTP/1.1" 127.0.0.1:8888
Destination: [127.0.0.1]:8888
Interface lo0 address [127.0.0.1]:0
Using interface lo0 to connect to [127.0.0.1]:8888
Ramped up to 1 connections.
Total data sent:     17.6 MiB (18411996 bytes)
Total data received: 24.0 MiB (25179388 bytes)
Bandwidth per channel: 34.844⇅ Mbps (4355.6 kBps)
Aggregate bandwidth: 20.127↓, 14.717↑ Mbps
Packet rate estimate: 47309.1↓, 1276.9↑ (1↓, 36↑ TCP MSS/op)
Message latency at percentiles: 583.1/597.5/601.9 ms (95/99/99.5%)
Test duration: 10.0082 s.
```

Reply:
I'd be interested to see the differences on Linux if it's not too hard. It's unlikely, but I wouldn't be surprised if allocations had more overhead on non-Apple platforms for some reason.

Reply:
@tanner0101 negligible difference on Linux too

```
### BEFORE (2 runs)
$ wrk http://127.0.0.1:8888/
Running 10s test @ http://127.0.0.1:8888/
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    76.19us  141.63us   9.45ms   99.69%
    Req/Sec    67.44k     3.77k   71.58k    96.50%
  1341906 requests in 10.00s, 66.55MB read
Requests/sec: 134187.61
Transfer/sec:      6.65MB
jweiss:~
$ wrk http://127.0.0.1:8888/
Running 10s test @ http://127.0.0.1:8888/
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    76.90us  181.85us  17.75ms   99.78%
    Req/Sec    67.09k     4.60k   77.69k    95.52%
  1341501 requests in 10.10s, 66.53MB read
Requests/sec: 132827.62
Transfer/sec:      6.59MB
jweiss:~

### AFTER (2 runs)
$ wrk http://127.0.0.1:8888/
Running 10s test @ http://127.0.0.1:8888/
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    78.88us  242.17us  19.57ms   99.63%
    Req/Sec    68.64k     4.61k   71.20k    95.52%
  1372871 requests in 10.10s, 68.08MB read
Requests/sec: 135934.67
Transfer/sec:      6.74MB
jweiss:~
$ wrk http://127.0.0.1:8888/
Running 10s test @ http://127.0.0.1:8888/
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    73.83us  112.64us   9.51ms   99.67%
    Req/Sec    68.66k     4.37k   72.27k    96.50%
  1366460 requests in 10.00s, 67.76MB read
Requests/sec: 136638.10
Transfer/sec:      6.78MB
```

Reply:
@tanner0101 and we'll be able to make it faster with #857

Reply:
because at the moment the code  path is just plain stupid ;). ByteBuffer -> String -> ByteBuffer -> String for every header :D

Reply:
Awesome, thanks for testing that :)